### DISCUSSION TITLE
Typescript useState<string> ReferenceError: string is not defined

### DISCUSSION THREAD

**Discussion by gsambrotta** [vercel/next.js#17818] [Help] [answered]
Hi there, 
I need your help because I already google so much and wrote also in SO but no luck so far!
premis: I'm new to typescrypt.

I'm using it in a project with Nextjs. All is going good except when I use types to define useState value.
For what I read, I should be able to do something like:

`const [emailValue, setEmailValue] = useState<string>('')`

But if I do so in my project, I get: `ReferenceError: string is undefined`.

For what I understood, typescript is not compiled properly and it prints `<string>` after compiling.
Indeed, I don't get an error on my IDE but just at runtime.

Does anyone have the same problem?
My typescript config are the default one from nextjs.
I have added react, node and react-dom types.

```
// package.json dpendencies
"dependencies": {
    "@types/classnames": "^2.2.10",
    "es6-promise": "4.2.8",
    "next": "9.5.3",
    "react": "16.13.1",
    "react-dom": "16.13.1",
    "typescript": "4.0.2"
  },
  "devDependencies": {
    "@babel/core": "~7.11.6",
    "@babel/plugin-transform-runtime": "~7.11.5",
    "@babel/preset-flow": "~7.10.4",
    "@storybook/addon-docs": "^6.0.21",
    "@storybook/react": "~6.0.21",
    "@types/enzyme": "~3.10.6",
    "@types/jest": "~26.0.14",
    "@types/node": "~14.10.3",
    "@types/react": "~16.9.49",
    "@types/react-dom": "~16.9.8",
    "@typescript-eslint/eslint-plugin": "~4.0.1",
    "@typescript-eslint/parser": "~4.0.1",
    "awesome-typescript-loader": "~5.2.1",
    "babel-eslint": "~10.1.0",
    "babel-jest": "~26.3.0",
    "babel-loader": "~8.1.0",
    "babel-plugin-transform-es2015-modules-commonjs": "~6.26.2",
    "classnames": "~2.2.6",
    "enzyme": "~3.11.0",
    "enzyme-adapter-react-16": "~1.15.4",
    "eslint": "~7.9.0",
    "eslint-config-standard": "~14.1.1",
    "eslint-config-standard-jsx": "~8.1.0",
    "eslint-plugin-import": "~2.22.1",
    "eslint-plugin-jest": "^24.1.0",
    "eslint-plugin-node": "~11.1.0",
    "eslint-plugin-promise": "~4.2.1",
    "eslint-plugin-react": "~7.21.3",
    "eslint-plugin-react-hooks": "~4.1.2",
    "eslint-plugin-standard": "^4.0.1",
    "fetch-jsonp": "~1.1.3",
    "husky": "~4.3.0",
    "jest": "~26.4.2",
    "lint-staged": "~10.3.0",
    "prettier": "~2.1.1",
    "react-docgen-typescript-loader": "~3.7.2",
    "ts-jest": "~26.3.0"
```

```
// tsconfig.json
// create by nextjs
{
  "compilerOptions": {
    "target": "es5",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": false,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve"
  },
  "include": [
    "next-env.d.ts",
    "**/*.ts",
    "**/*.tsx"
  ],
  "exclude": [
    "node_modules"
  ]
}

```

**Comment by jamesmosier** [comment, marked as answer]
Hi @gsambrotta. Do you have `typescript` installed in `devDependencies`? Also what does your tsconfig.json look like?

**Reply by gsambrotta** [reply]
Thank you for your replay @jamesmosier!
Yes, I do have typescrypt. As I wrote I can use any other Typescrypt/react types (FC, define props types, ..)
And my tsconfig is the default one created by Nextjs
Anyway, I added my dependencies and typescript config in the post so is more straight forward.

**Reply by fabarea** [reply]
I am facing the same issue. TypeScript is well installed as a dev dependency.

**Reply by jamesmosier** [reply]
Do you have a repository you could share?

**Reply by fabarea** [reply]
I found the / my issue. I had a `babel.config.json` which was overriding the default configuration. Renaming / deleting the file fixed the issue.

**Reply by gsambrotta** [reply]
Thank you @fabarea !!! That's it! 
I don't even remember when I put all that stuff in my `babel.config.json` 🤔  🤔 

Anyway, I still need it to run my test in Jest but I figured out I can simply write

```
// babel.config.json
{
  "presets": ["next/babel"]
}
```

And works pretty well
